File: BotTemplate/BotTemplate-Java/src/main/java/io/github/freya022/bot/ReadyListener.java
package io.github.freya022.bot;

import com.freya02.botcommands.api.Logging;
import com.freya02.botcommands.api.core.annotations.BEventListener;
import com.freya02.botcommands.api.core.service.annotations.BService;
import net.dv8tion.jda.api.JDA;
import net.dv8tion.jda.api.entities.Guild;
import net.dv8tion.jda.api.events.session.ReadyEvent;
import org.slf4j.Logger;

@BService
public class ReadyListener {
    private static final Logger LOGGER = Logging.getLogger();

    @BEventListener
    public void onReady(ReadyEvent event) {
        final JDA jda = event.getJDA();

        LOGGER.info("Bot connected as {}", jda.getSelfUser().getName());
        LOGGER.info("The bot is present on these guilds :");
        for (Guild guild : jda.getGuildCache()) {
            LOGGER.info("\t- {} ({})", guild.getName(), guild.getId());
        }
    }
}

File: src/examples/kotlin/io/github/freya022/bot/ReadyListener.kt
package io.github.freya022.bot

import io.github.freya022.botcommands.api.core.annotations.BEventListener
import io.github.freya022.botcommands.api.core.service.annotations.BService
import io.github.oshai.kotlinlogging.KotlinLogging
import kotlinx.coroutines.delay
import net.dv8tion.jda.api.events.session.ReadyEvent
import kotlin.time.Duration.Companion.milliseconds

private val logger = KotlinLogging.logger { }

// You can optionally have a name to differentiate between multiple instance of your services
@BService(name = "myReadyListener")
class ReadyListener {
    @BEventListener(priority = 1)
    fun onReadyFirst(event: ReadyEvent) {
        logger.info { "First handling of ReadyEvent" }

        val jda = event.jda

        //Print some information about the bot
        logger.info { "Bot connected as ${jda.selfUser.name}" }
        logger.info { "The bot is present on these guilds :" }
        for (guild in jda.guildCache) {
            logger.info { "\t- ${guild.name} (${guild.id})" }
        }
    }

    // Executes after the above listener, but doesn't prevent the listener below from running
    @BEventListener(priority = 0, async = true)
    suspend fun onReadyAsync(event: ReadyEvent) {
        logger.info { "(Before) Async handling of ReadyEvent" }
        delay(200.milliseconds)
        logger.info { "(After) Async handling of ReadyEvent" }
    }

    // Executes after the above listener
    @BEventListener(priority = -1)
    fun onReadyLast(event: ReadyEvent) {
        logger.info { "Last handling of ReadyEvent" }
    }
}